src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java

```diff
@@ -0,0 +1,213 @@
+        }
+        for (Car car: carList) {
+            car.setDrivers(getDriversForCar(car.getId()));
+        }
+        return carList;
+    }
+
+    @Override
+    public Car update(Car car) {
+        String query = "UPDATE cars SET model = ?, manufacturer_id = ? "
+                + "WHERE id = ? AND is_deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setString(1, car.getModel());
+            statement.setObject(2, car.getManufacturer().getId());
+            statement.setLong(3, car.getId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't update "
+                    + car + " in DB.", e);
+        }
+        removeDriversFromCar(car.getId());
+        addDriversToCar(car);
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE cars SET is_deleted = TRUE WHERE id = ?";
+        boolean deleted;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            deleted = statement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't delete car with id " + id, e);
+        }
+        removeDriversFromCar(id);
+        return deleted;
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String getCarsByDriverQuery = "SELECT id, model, manufacturer_id "
+                + "FROM cars c JOIN cars_drivers cd ON c.id = cd.car_id "
+                + "WHERE cd.driver_id = ? AND is_deleted = FALSE;";
+        List<Car> carList = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(getCarsByDriverQuery)) {
+            statement.setLong(1, driverId);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                carList.add(getCarFromResultSet(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get all "
+                    + "cars with driver id " + driverId, e);
+        }
+        for (Car car: carList) {
+            car.setDrivers(getDriversForCar(car.getId()));
+        }
+        return carList;
+    }
+
+    private Car getCarFromResultSet(ResultSet resultSet) throws SQLException {
+        Car car = new Car();
+        car.setId(resultSet.getObject("car_id", Long.class));
+        car.setModel(resultSet.getString("model"));
+        Manufacturer manufacturer = manufacturerService.get(resultSet
+                .getObject("manufacturer_id", Long.class));
```

Comment:
nice try, but no 😄 
![image](https://user-images.githubusercontent.com/117590814/213980104-b42a91e1-6dea-4668-a633-7203995b43d3.png)
^ from common mistakes
also using service methods in dao is just wrong
you supposed to do double join in method getAllByDriver
SELECT ... from car join cars_drivers on ... join manufactures on ... where ...